From 633ce3fa477524c794828faca56e786d6c2e1da4 Mon Sep 17 00:00:00 2001
From: Sudheesh Mavila <sudheesh.mavila@amd.com>
Date: Tue, 12 Jan 2021 15:11:03 +0530
Subject: [PATCH 32/48] amd-xgbe: Fix for Network fluctuations

BEL SFP, when connected to amd-xgbe shows frequent link down and up events on dmesg.
Refer ticket EMBDEV-8951. This fix avoids the frquent link up/down issue.

Signed-off-by: Sudheesh Mavila <sudheesh.mavila@amd.com>
---
 drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c b/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
index 765479904723..371b4c91e289 100644
--- a/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
+++ b/drivers/net/ethernet/amd/xgbe/xgbe-phy-v2.c
@@ -1049,6 +1049,18 @@ static int xgbe_phy_find_phy_device(struct xgbe_prv_data *pdata)
 	}
 	phy_data->phydev = phydev;
 
+	switch (phy_data->port_mode) {
+	case XGBE_PORT_MODE_SFP:
+		/*  reset the sfp phy  EMBDEV-8951 */
+		if(phydev)
+			genphy_soft_reset(phydev);
+		else
+			netdev_err(pdata->netdev, "phy reset failed\n");
+		break;
+	default:
+		break;
+	}
+
 	xgbe_phy_external_phy_quirks(pdata);
 
 	linkmode_and(phydev->advertising, phydev->advertising,
-- 
2.27.0

